src/services/measure-handler.ts

```diff
@@ -488,76 +540,80 @@ export const validateMeasureLookupTable = async (
 async function getMeasurePreviewWithoutExtractor(
   dataset: Dataset,
   measure: Measure,
-  quack: Database,
-  tableName: string
+  revision: Revision
 ): Promise<ViewDTO> {
-  const preview = await quack.all(
-    `SELECT DISTINCT "${measure.factTableColumn}" FROM ${tableName} ORDER BY "${measure.factTableColumn}" ASC LIMIT ${sampleSize};`
-  );
-  const tableHeaders = Object.keys(preview[0]);
-  const dataArray = preview.map((row) => Object.values(row));
-  const currentDataset = await DatasetRepository.getById(dataset.id);
-  const headers: CSVHeader[] = [];
-  for (let i = 0; i < tableHeaders.length; i++) {
-    headers.push({
-      index: i,
-      name: tableHeaders[i],
-      source_type: FactTableColumnType.Unknown
-    });
+  const quack = await duckdb();
+  await linkToPostgres(quack, revision.id, false);
+  try {
+    const preview = await quack.all(
+      pgformat(
+        'SELECT DISTINCT %I FROM %I ORDER BY %I ASC LIMIT %L;',
+        measure.factTableColumn,
+        'fact_table',
+        measure.factTableColumn,
+        sampleSize
+      )
+    );
+
+    const tableHeaders = Object.keys(preview[0]);
+    const dataArray = preview.map((row) => Object.values(row));
+    const currentDataset = await DatasetRepository.getById(dataset.id);
+    const headers: CSVHeader[] = [];
+    for (let i = 0; i < tableHeaders.length; i++) {
+      headers.push({
+        index: i,
+        name: tableHeaders[i],
+        source_type: FactTableColumnType.Unknown
+      });
+    }
```

Comment:
We do this same map in at least a dozen places, it should have a helper function by now.

There's `tableDataToViewTable()` but it looks like it has an additional wrapper object.